engine/routes.py

```diff
@@ -49,13 +52,19 @@ def handle_exception(error):
 
 @app.before_request
 def log_request():
-    data = request.get_json()
-    request_id = data.get('requestUUID')
-    streams = data.get('streams')
-    if log.isEnabledFor(logging.DEBUG):
-        log.debug('<%s> Incoming request url=%r data=%r', request_id, request.url, data)
+    if request.endpoint == 'version':
+        if log.isEnabledFor(logging.DEBUG):
+            log.debug('Incoming request url=%r', request.url)
+        else:
+            log.info('Handling request to %r', request.url)
     else:
-        log.info('<%s> Handling request to %r - %r', request_id, request.url, streams)
+        data = request.get_json()
```

Comment:
Here we want to have a different output when debug is turned on, so it makes sense to check the logging level, however, it's safer to provide a default to avoid dictionary lookup issues:

```python
data = request.get_json()
request_id = data.get('requestUUID', None)
if log.isEnabledFor(logging.DEBUG):
    log.debug('<%s> Incoming request url=%r data=%r', request_id, request.url, data)
else:
    streams = data.get('streams', None)
    log.debug('<%s> Incoming request url=%r data=%r', request_id, request.url, streams)
```
Consider if this is really what we want to do. If this is performant code, getting rid of the if/else block and just doing both the logging statements is optimal. 

```python
data = request.get_json()
request_id = data.get('requestUUID', None)
log.debug('<%s> Incoming request url=%r data=%r', request_id, request.url, data)
streams = data.get('streams', None)
log.info('<%s> Incoming request url=%r data=%r', request_id, request.url, streams)
```

Reply:
When the log level is set to debug, the redundant logging will be a little annoying. That being said, its debugging output and should be used rarely. This code will run every time Stream Engine receives a request, so I think the performance gain of doing both log statements is worth the redundancy in debugging logging. Removing the if/else...